### PR TITLE
Resolved Trigger deadLetterSink is available in the status

### LINKED ISSUE #1086: [Tracking knative/eventing#5363] Resolved Trigger deadLetterSink is not available in status
**Problem**

- https://github.com/knative/eventing/issues/5363

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
Event Consumer

**Exit Criteria**
- Resolved Trigger deadLetterSink is available in status 
- Existing tests pass

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
0.1

**Additional context (optional)**
- Eventing implementation https://github.com/knative/eventing/pull/5551

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes #1086

## Proposed Changes

- Resolved Trigger deadLetterSink is available in the status
- Upgrade eventing by running `go get -u knative.dev/eventing@main && ./hack/update-deps.sh` 
 

**Release Note**

```release-note
The Trigger status contains the resolved URL of the configured dead letter sink.
```

/kind enhancement

### CODE REVIEW

File: control-plane/pkg/core/config/utils.go [hunk 1]
```diff
@@ -136,3 +136,41 @@ func DurationMillisFromISO8601String(durationStr *string, defaultDurationMillis
 func IncrementContractGeneration(ct *contract.Contract) {
 	ct.Generation = (ct.Generation + 1) % (math.MaxUint64 - 1)
 }
+
+// MergeEgressConfig merges the 2 given egress configs into one egress config prioritizing e0 values.
+func MergeEgressConfig(e0, e1 *contract.EgressConfig) *contract.EgressConfig {
+	if e0 == nil {
+		return e1
```
Comment (matzew) [inline control-plane/pkg/core/config/utils.go:143]: what if `e1` is nil? :smile:

File: control-plane/pkg/core/config/utils_test.go [hunk 2]
```diff
@@ -346,3 +349,133 @@ func TestEgressConfigFromDelivery(t *testing.T) {
...
+				DeadLetter:   "e0",
+				Retry:        43,
+				Timeout:      100,
+				BackoffDelay: 4001,
+			},
+		},
+		{
+			name: "e0 backoff policy priority",
+			e0: &contract.EgressConfig{
+				DeadLetter:    "e0",
```
Comment (matzew) [inline control-plane/pkg/core/config/utils_test.go:445]: what if this has `Retry`?

Comment (matzew) [reply control-plane/pkg/core/config/utils_test.go:445]: it should have prio over `e1`, right ?

**Comment by matzew** [comment]
a bit of a nit here, see my comments.

Is it possible to also update/rebase this PR, inclduing the upgrade? So we can close 1088 when this is in?

**Comment by matzew** [comment]
@pierDipi or should we merge this, and just wait for the automated updater (fine with me too)

File: control-plane/pkg/core/config/utils.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/core/config/utils.go:143]: That is fine.

File: control-plane/pkg/core/config/utils_test.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/core/config/utils_test.go:445]: Right, let me add another test with the retry parameter

**Comment by pierDipi** [comment]
Yes, we can merge this, and wait for the bot to rebase https://github.com/knative-sandbox/eventing-kafka-broker/pull/1088

File: control-plane/pkg/core/config/utils_test.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/core/config/utils_test.go:445]: Done!

File: control-plane/pkg/core/config/utils.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/core/config/utils.go:143]: In that case we have `MergeEgressConfig(nil, nil) = nil`

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by pierDipi** [comment]
Webhook EOF error
/retest

**Pull request merged**